src/pages/KakaoRedirect.tsx

```diff
@@ -0,0 +1,46 @@
+import { useNavigate } from 'react-router'
+import { useQuery } from '@tanstack/react-query'
+import { useEffect } from 'react'
+import axiosInstance from '../api/axiosInstance'
+
+const fetchLogin = async (code: string | null): Promise<string> => {
+  const { data } = await axiosInstance.get(`/api/oauth/login`, {
+    params: {
+      code,
+    },
+  })
+  return data
+}
+
+const KakaoRedirect = () => {
+  const navigate = useNavigate()
+  const code = new URL(document.location.toString()).searchParams.get('code')
+
+  const {
+    data: tokenData,
+    isSuccess,
+    isLoading,
+    isError,
+    error: queryError,
+  } = useQuery({
+    queryKey: ['kakaoLogin', code],
+    queryFn: () => fetchLogin(code),
+    enabled: !!code,
+    retry: 1,
+  })
+
+  useEffect(() => {
+    if (isSuccess) {
+      localStorage.setItem('authToken', tokenData)
+      navigate('/')
+    }
+  })
```

Comment:
```suggestion
  }, [isSuccess, tokenData, navigate])
```

의존성 배열이 없다면 리렌더링 될 떄 마다 실행될거라 추가해주는게 좋겠습니다. 또한 eslint에서 eslint-plugin-react-hooks를 활용해 react-hooks/exhaustive-deps 규칙을 활성화하면 자동으로 검출해줍니다.